Tests/MathLib/mechanics_invariants.cpp

```diff
@@ -0,0 +1,286 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/LICENSE.txt
+ */
+
+#include <gtest/gtest.h>
+#include <autocheck/autocheck.hpp>
+
+#include "MathLib/mechanics_invariants.h"
+
+#include "AutoCheckTools.h"
+
+using namespace MathLib;
+namespace ac = autocheck;
+
+template <int Size>
+using KelvinVector = Eigen::Matrix<double, Size, 1>;
+
+template <int KelvinVectorSize>
+struct KelvinToTensorSize;
+template <>
+struct KelvinToTensorSize<1>
+{
+    static const int value = 1;
+};
+template <>
+struct KelvinToTensorSize<4>
+{
+    static const int value = 2;
+};
+template <>
+struct KelvinToTensorSize<6>
+{
+    static const int value = 3;
+};
+
+template <int Size>
+Eigen::Matrix<double, KelvinToTensorSize<Size>::value,
+              KelvinToTensorSize<Size>::value>
+kelvinToTensor(KelvinVector<Size> const& v);
+
+template <>
+Eigen::Matrix<double, KelvinToTensorSize<1>::value,
+              KelvinToTensorSize<1>::value>
+    kelvinToTensor(KelvinVector<1> const& v)
+{
+    Eigen::Matrix<double, 1, 1> m;
+    m.coeffRef(0,0) = v[0];
+    return m;
+}
+
+template <>
+Eigen::Matrix<double, KelvinToTensorSize<4>::value,
+              KelvinToTensorSize<4>::value>
+    kelvinToTensor(KelvinVector<4> const& v)
+{
+    Eigen::Matrix<double, KelvinToTensorSize<4>::value,
+                  KelvinToTensorSize<4>::value>
+        m;
+    // The z-component is ignored, otherwise the matrix must be 3x3.
```

Comment:
Maybe then the matrix better should be 3x3? I guess it would be nice if `K2T(T2K(A)) == A`